batch/batch/worker/worker.py

```diff
@@ -2323,11 +2320,14 @@ async def run(self):
                     mjs_fut.cancel()
 
     async def cleanup(self):
+        assert CLOUD_WORKER_API
         assert self.worker
         assert self.worker.file_store is not None
         assert self.worker.fs
         assert self.jvm
 
+        await CLOUD_WORKER_API.remove_user_credentials(self.hail_identity)
```

Comment:
~~I'd expect to need some kind of reference counting; why don't we need that? Couldn't multiple jobs share creds?~~

Heh. I think I wouldn't call it remove. Not sure what the right name is but remove feels wrong. Maybe "release"?